internal/config/loader.go

```diff
@@ -38,10 +38,38 @@ var (
 // whether the service registers itself the registry. The profile and confDir
 // are used to locate the local TOML config file.
 func LoadConfig(useRegistry bool, profile string, confDir string) (config *common.Config, err error) {
-	fmt.Fprintf(os.Stdout, "Init: useRegistry: %v profile: %s confDir: %s\n",
+	_, _ = fmt.Fprintf(os.Stdout, "Init: useRegistry: %v profile: %s confDir: %s\n",
 		useRegistry, profile, confDir)
-	confName := "configuration.toml"
+	config, err = loadConfigFromFile(profile, confDir)
 
+	var registryMsg string
+	if useRegistry {
+		registryMsg = "Register in registry...\n"
+		RegistryClient, err = GetRegistryClient(common.ServiceName, config)
+		if err != nil {
+			return nil, err
+		}
+		if RegistryClient.CheckConfigExistence() {
+			config, err = RegistryClient.LoadConfig(config)
+			if err != nil {
+				_, _ = fmt.Fprintf(os.Stderr, "Load config from Consul failed... %v \n", err)
```

Comment:
Should be an error is expecting to load from Consul, aka Registry, and it doesn't contain the configuration. Otherwise it silently fails to do what was asked.

Reply:
@lenny-intel you are right.  I forgot to return err.

Reply:
@cloudxxx8 , Ok now I see that you are pushing the config into Registry if not found in the Registry. I'll add that to my implementation using the new go-mod-registry.